Fix: avoid multiple pending txs with the same nonce

katspaugh opened the pull request:
## What it solves

Resolves #1667

## How this PR fixes it

Disallows immediately executing a newly created tx w/o signatures if there are already other pending transactions.

## How to test
In a 1/X Safe, create and immediately execute a tx. Remember its nonce.
Quickly create another tx. Notice that it will have the same nonce as the previous tx, but the Execute option will be disabled.

iamacook commented on src/components/tx/SignOrExecuteForm/index.tsx:
```diff
@@ -62,11 +63,12 @@ const SignOrExecuteForm = ({
   const isOwner = useIsSafeOwner()
   const provider = useWeb3()
   const currentChain = useCurrentChain()
+  const hasPending = useHasPendingTxs()
```
We should also check that the pending transactions are on the current chain.

francovenica commented:
Didn't worked for me. 
Created the a tx and executed it right away
Quickly created a new tx. This one had the same nonce as the one I was going to execute but with no "Execute now" checkbox in the modal. Queued it anyways
Waited for the first tx to finish to execute
The result is 2 tx with the same nonce, one in history and one in queue. The one in history is labeled as "untrusted" and theone in queue can still be executed (or at least you can attempt to)

Note: I had to speed up the 1st tx cuz it was taking a long time to execute, but that just the network being slow.
Note2: I had to refresh the history tab manually for the executed tx to show up

The safe:
https://fix_nonce--webcore.review-web-core.5afe.dev/transactions/queue?safe=gor:0x24F8542Da0C5f267c177a224270442C73804EE80
![tx nonces](https://user-images.githubusercontent.com/4503659/220386833-e1dbdd07-0f1a-49be-8f33-5bb6cf8b1eb3.gif)

katspaugh commented:
That's expected. You can replace that tx.

francovenica commented:
After a while (1h +) the tx with nonce 150 that was in queue dissapeared on its own

katspaugh commented:
It disappeared because the other tx with the same nonce got indexed and replaced the queued one. So it's correct.

I've added a feature to display untrusted pending transactions on top of the regular queue. This should make things more transparent.

<img width="1240" alt="Screenshot 2023-02-22 at 09 34 08" src="https://user-images.githubusercontent.com/381895/220566293-b0c71c08-f7c4-4ac2-b352-b3b21aca2a97.png">

<img width="1272" alt="Screenshot 2023-02-22 at 09 34 17" src="https://user-images.githubusercontent.com/381895/220566338-f095f31d-d6e7-4ba9-a18c-1e7ad88068e1.png">

francovenica commented:
Well, I tried again in Poly ( goerli is pretty slow these days) and the I create 2 tx quickly. The first one is executing/indexing, and the 2nd one, with the same nonce as the 1st one, was cancelled right away

I didn't had the issue of having 2 tx with the same nonce in the history and queue again.
